### PR TITLE
Local group creation for enduser assignment - fix #1226 #1624 #1591

### LINKED ISSUE #1226: Error creating controller. Please check Docker is running using `docker ps`.
There are logs:

2024-04-19 21:40:10 INFO:     Started server process [1]
2024-04-19 21:40:10 INFO:     Waiting for application startup.
2024-04-19 21:40:10 INFO:     Application startup complete.
2024-04-19 21:40:10 INFO:     Uvicorn running on http://0.0.0.0:3000 (Press CTRL+C to quit)
2024-04-19 21:40:15 INFO:     ('172.17.0.1', 45180) - "WebSocket /ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzaWQiOiI1NzNjMGQzNi1hMzdlLTRkYWQtYjdjYy0xMzY3YTU3ZGM5NDMifQ.XOJV2GTeQpwaXW3v7vr7sPK_1CPXv4fxefY69GA1JfE" [accepted]
2024-04-19 21:40:15 INFO:     connection open
2024-04-19 21:40:15 Starting loop_recv for sid: 573c0d36-a37e-4dad-b7cc-1367a57dc943
2024-04-19 21:40:15 INFO:     172.17.0.1:45202 - "GET /api/messages/total HTTP/1.1" 200 OK
2024-04-19 21:40:15 INFO:     172.17.0.1:45194 - "GET /api/configurations HTTP/1.1" 200 OK
2024-04-19 21:40:15 13:40:15 - opendevin:ERROR: exec_box.py:55 - Please check Docker is running using `docker ps`.
2024-04-19 21:40:15 13:40:15 - opendevin:ERROR: agent.py:127 - Error creating controller: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:40:15 Traceback (most recent call last):
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 793, in urlopen
2024-04-19 21:40:15     response = self._make_request(
2024-04-19 21:40:15                ^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 496, in _make_request
2024-04-19 21:40:15     conn.request(
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connection.py", line 400, in request
2024-04-19 21:40:15     self.endheaders()
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1331, in endheaders
2024-04-19 21:40:15     self._send_output(message_body, encode_chunked=encode_chunked)
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1091, in _send_output
2024-04-19 21:40:15     self.send(msg)
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1035, in send
2024-04-19 21:40:15     self.connect()
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/transport/unixconn.py", line 27, in connect
2024-04-19 21:40:15     sock.connect(self.unix_socket)
2024-04-19 21:40:15 FileNotFoundError: [Errno 2] No such file or directory
2024-04-19 21:40:15 
2024-04-19 21:40:15 During handling of the above exception, another exception occurred:
2024-04-19 21:40:15 
2024-04-19 21:40:15 Traceback (most recent call last):
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/requests/adapters.py", line 486, in send
2024-04-19 21:40:15     resp = conn.urlopen(
2024-04-19 21:40:15            ^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 847, in urlopen
2024-04-19 21:40:15     retries = retries.increment(
2024-04-19 21:40:15               ^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/util/retry.py", line 470, in increment
2024-04-19 21:40:15     raise reraise(type(error), error, _stacktrace)
2024-04-19 21:40:15           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/util/util.py", line 38, in reraise
2024-04-19 21:40:15     raise value.with_traceback(tb)
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 793, in urlopen
2024-04-19 21:40:15     response = self._make_request(
2024-04-19 21:40:15                ^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 496, in _make_request
2024-04-19 21:40:15     conn.request(
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/urllib3/connection.py", line 400, in request
2024-04-19 21:40:15     self.endheaders()
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1331, in endheaders
2024-04-19 21:40:15     self._send_output(message_body, encode_chunked=encode_chunked)
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1091, in _send_output
2024-04-19 21:40:15     self.send(msg)
2024-04-19 21:40:15   File "/usr/local/lib/python3.12/http/client.py", line 1035, in send
2024-04-19 21:40:15     self.connect()
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/transport/unixconn.py", line 27, in connect
2024-04-19 21:40:15     sock.connect(self.unix_socket)
2024-04-19 21:40:15 urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:40:15 
2024-04-19 21:40:15 During handling of the above exception, another exception occurred:
2024-04-19 21:40:15 
2024-04-19 21:40:15 Traceback (most recent call last):
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 213, in _retrieve_server_version
2024-04-19 21:40:15     return self.version(api_version=False)["ApiVersion"]
2024-04-19 21:40:15            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/api/daemon.py", line 181, in version
2024-04-19 21:40:15     return self._result(self._get(url), json=True)
2024-04-19 21:40:15                         ^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/utils/decorators.py", line 44, in inner
2024-04-19 21:40:15     return f(self, *args, **kwargs)
2024-04-19 21:40:15            ^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 236, in _get
2024-04-19 21:40:15     return self.get(url, **self._set_request_timeout(kwargs))
2024-04-19 21:40:15            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 602, in get
2024-04-19 21:40:15     return self.request("GET", url, **kwargs)
2024-04-19 21:40:15            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
2024-04-19 21:40:15     resp = self.send(prep, **send_kwargs)
2024-04-19 21:40:15            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
2024-04-19 21:40:15     r = adapter.send(request, **kwargs)
2024-04-19 21:40:15         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/requests/adapters.py", line 501, in send
2024-04-19 21:40:15     raise ConnectionError(err, request=request)
2024-04-19 21:40:15 requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:40:15 
2024-04-19 21:40:15 The above exception was the direct cause of the following exception:
2024-04-19 21:40:15 
2024-04-19 21:40:15 Traceback (most recent call last):
2024-04-19 21:40:15   File "/app/opendevin/server/agent/agent.py", line 118, in create_controller
2024-04-19 21:40:15     self.controller = AgentController(
2024-04-19 21:40:15                       ^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/opendevin/controller/agent_controller.py", line 51, in __init__
2024-04-19 21:40:15     self.command_manager = CommandManager(self.id, container_image)
2024-04-19 21:40:15                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/opendevin/controller/command_manager.py", line 20, in __init__
2024-04-19 21:40:15     self.shell = DockerExecBox(
2024-04-19 21:40:15                  ^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/opendevin/sandbox/exec_box.py", line 57, in __init__
2024-04-19 21:40:15     raise ex
2024-04-19 21:40:15   File "/app/opendevin/sandbox/exec_box.py", line 53, in __init__
2024-04-19 21:40:15     self.docker_client = docker.from_env()
2024-04-19 21:40:15                          ^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/client.py", line 94, in from_env
2024-04-19 21:40:15     return cls(
2024-04-19 21:40:15            ^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/client.py", line 45, in __init__
2024-04-19 21:40:15     self.api = APIClient(*args, **kwargs)
2024-04-19 21:40:15                ^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 197, in __init__
2024-04-19 21:40:15     self._version = self._retrieve_server_version()
2024-04-19 21:40:15                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:40:15   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 220, in _retrieve_server_version
2024-04-19 21:40:15     raise DockerException(
2024-04-19 21:40:15 docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:44:53 INFO:     172.17.0.1:54070 - "GET / HTTP/1.1" 307 Temporary Redirect
2024-04-19 21:44:53 INFO:     ('172.17.0.1', 54084) - "WebSocket /ws?token=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJzaWQiOiI1NzNjMGQzNi1hMzdlLTRkYWQtYjdjYy0xMzY3YTU3ZGM5NDMifQ.XOJV2GTeQpwaXW3v7vr7sPK_1CPXv4fxefY69GA1JfE" [accepted]
2024-04-19 21:44:53 INFO:     connection open
2024-04-19 21:44:53 Starting loop_recv for sid: 573c0d36-a37e-4dad-b7cc-1367a57dc943
2024-04-19 21:44:53 INFO:     172.17.0.1:54070 - "GET /locales/zh/translation.json HTTP/1.1" 404 Not Found
2024-04-19 21:44:53 INFO:     172.17.0.1:54070 - "GET /api/refresh-files HTTP/1.1" 200 OK
2024-04-19 21:44:53 INFO:     172.17.0.1:54072 - "GET /api/messages/total HTTP/1.1" 200 OK
2024-04-19 21:44:53 INFO:     172.17.0.1:54070 - "GET /api/configurations HTTP/1.1" 200 OK
2024-04-19 21:44:53 13:44:53 - opendevin:ERROR: exec_box.py:55 - Please check Docker is running using `docker ps`.
2024-04-19 21:44:53 13:44:53 - opendevin:ERROR: agent.py:127 - Error creating controller: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:44:53 Traceback (most recent call last):
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 793, in urlopen
2024-04-19 21:44:53     response = self._make_request(
2024-04-19 21:44:53                ^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 496, in _make_request
2024-04-19 21:44:53     conn.request(
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connection.py", line 400, in request
2024-04-19 21:44:53     self.endheaders()
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1331, in endheaders
2024-04-19 21:44:53     self._send_output(message_body, encode_chunked=encode_chunked)
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1091, in _send_output
2024-04-19 21:44:53     self.send(msg)
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1035, in send
2024-04-19 21:44:53     self.connect()
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/transport/unixconn.py", line 27, in connect
2024-04-19 21:44:53     sock.connect(self.unix_socket)
2024-04-19 21:44:53 FileNotFoundError: [Errno 2] No such file or directory
2024-04-19 21:44:53 
2024-04-19 21:44:53 During handling of the above exception, another exception occurred:
2024-04-19 21:44:53 
2024-04-19 21:44:53 Traceback (most recent call last):
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/requests/adapters.py", line 486, in send
2024-04-19 21:44:53     resp = conn.urlopen(
2024-04-19 21:44:53            ^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 847, in urlopen
2024-04-19 21:44:53     retries = retries.increment(
2024-04-19 21:44:53               ^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/util/retry.py", line 470, in increment
2024-04-19 21:44:53     raise reraise(type(error), error, _stacktrace)
2024-04-19 21:44:53           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/util/util.py", line 38, in reraise
2024-04-19 21:44:53     raise value.with_traceback(tb)
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 793, in urlopen
2024-04-19 21:44:53     response = self._make_request(
2024-04-19 21:44:53                ^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connectionpool.py", line 496, in _make_request
2024-04-19 21:44:53     conn.request(
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/urllib3/connection.py", line 400, in request
2024-04-19 21:44:53     self.endheaders()
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1331, in endheaders
2024-04-19 21:44:53     self._send_output(message_body, encode_chunked=encode_chunked)
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1091, in _send_output
2024-04-19 21:44:53     self.send(msg)
2024-04-19 21:44:53   File "/usr/local/lib/python3.12/http/client.py", line 1035, in send
2024-04-19 21:44:53     self.connect()
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/transport/unixconn.py", line 27, in connect
2024-04-19 21:44:53     sock.connect(self.unix_socket)
2024-04-19 21:44:53 urllib3.exceptions.ProtocolError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:44:53 
2024-04-19 21:44:53 During handling of the above exception, another exception occurred:
2024-04-19 21:44:53 
2024-04-19 21:44:53 Traceback (most recent call last):
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 213, in _retrieve_server_version
2024-04-19 21:44:53     return self.version(api_version=False)["ApiVersion"]
2024-04-19 21:44:53            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/api/daemon.py", line 181, in version
2024-04-19 21:44:53     return self._result(self._get(url), json=True)
2024-04-19 21:44:53                         ^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/utils/decorators.py", line 44, in inner
2024-04-19 21:44:53     return f(self, *args, **kwargs)
2024-04-19 21:44:53            ^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 236, in _get
2024-04-19 21:44:53     return self.get(url, **self._set_request_timeout(kwargs))
2024-04-19 21:44:53            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 602, in get
2024-04-19 21:44:53     return self.request("GET", url, **kwargs)
2024-04-19 21:44:53            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 589, in request
2024-04-19 21:44:53     resp = self.send(prep, **send_kwargs)
2024-04-19 21:44:53            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/requests/sessions.py", line 703, in send
2024-04-19 21:44:53     r = adapter.send(request, **kwargs)
2024-04-19 21:44:53         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/requests/adapters.py", line 501, in send
2024-04-19 21:44:53     raise ConnectionError(err, request=request)
2024-04-19 21:44:53 requests.exceptions.ConnectionError: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:44:53 
2024-04-19 21:44:53 The above exception was the direct cause of the following exception:
2024-04-19 21:44:53 
2024-04-19 21:44:53 Traceback (most recent call last):
2024-04-19 21:44:53   File "/app/opendevin/server/agent/agent.py", line 118, in create_controller
2024-04-19 21:44:53     self.controller = AgentController(
2024-04-19 21:44:53                       ^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/opendevin/controller/agent_controller.py", line 51, in __init__
2024-04-19 21:44:53     self.command_manager = CommandManager(self.id, container_image)
2024-04-19 21:44:53                            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/opendevin/controller/command_manager.py", line 20, in __init__
2024-04-19 21:44:53     self.shell = DockerExecBox(
2024-04-19 21:44:53                  ^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/opendevin/sandbox/exec_box.py", line 57, in __init__
2024-04-19 21:44:53     raise ex
2024-04-19 21:44:53   File "/app/opendevin/sandbox/exec_box.py", line 53, in __init__
2024-04-19 21:44:53     self.docker_client = docker.from_env()
2024-04-19 21:44:53                          ^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/client.py", line 94, in from_env
2024-04-19 21:44:53     return cls(
2024-04-19 21:44:53            ^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/client.py", line 45, in __init__
2024-04-19 21:44:53     self.api = APIClient(*args, **kwargs)
2024-04-19 21:44:53                ^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 197, in __init__
2024-04-19 21:44:53     self._version = self._retrieve_server_version()
2024-04-19 21:44:53                     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
2024-04-19 21:44:53   File "/app/.venv/lib/python3.12/site-packages/docker/api/client.py", line 220, in _retrieve_server_version
2024-04-19 21:44:53     raise DockerException(
2024-04-19 21:44:53 docker.errors.DockerException: Error while fetching server API version: ('Connection aborted.', FileNotFoundError(2, 'No such file or directory'))
2024-04-19 21:44:55 INFO:     172.17.0.1:54070 - "GET /api/messages HTTP/1.1" 200 OK
2024-04-19 21:45:06 13:45:06 - opendevin:INFO: session.py:39 - WebSocket disconnected, sid: 573c0d36-a37e-4dad-b7cc-1367a57dc943
2024-04-19 21:45:06 INFO:     connection closed
——————————————
Already set the exports;

I used the under command:
docker run \
    -e LLM_API_KEY \
    -e WORKSPACE_MOUNT_PATH=$WORKSPACE_DIR \
    -v $WORKSPACE_DIR:/opt/workspace_base \
    -v /var/run/docker.sock:/var/run/docker.sock \
    -p 3000:3000 \
    -e SANDBOX_TYPE=exec \
    ghcr.io/opendevin/opendevin:0.3.1
    
what's the problem?

### PR DESCRIPTION
**Pull request by imkebe**
Lack of group caused `usermod` command to fail, hence no permissions resulting from group membership - docker socket access.

### CODE REVIEW

**Comment by Umpire2018** [comment]
LGTM.

But a suggestion, you can add linked issues to PR comment, like following.

image

**Comment by imkebe** [comment]
@Umpire2018 Sure. However I would expect someone to test it for different enviroments and then decide if can be closed. Ref
#1226 
#1624 
#1591

**Review by iFurySt — APPROVED** [review]

File: containers/app/entrypoint.sh [hunk 1]
```diff
@@ -33,8 +33,11 @@ mv /home/opendevin/.cache/ms-playwright/ /home/enduser/.cache/
 
 # get the user group of /var/run/docker.sock and set opendevin to that group
 DOCKER_SOCKET_GID=$(stat -c '%g' /var/run/docker.sock)
-echo "Docker socket group id: $DOCKER_SOCKET_GID"
-usermod -aG $DOCKER_SOCKET_GID enduser
+echo "Docker socket group id: $DOCKER_SOCKET_GID."
+echo "Creating docker group with fetched gid."
+groupadd -g $DOCKER_SOCKET_GID docker
```
Comment (rbren) [inline containers/app/entrypoint.sh:38]: do we need to check if the group exists first?

**Review by rbren — CHANGES_REQUESTED** [review]
I think we need to be a little more careful here.

This is what I get when running:
```
Creating docker group with fetched gid.
groupadd: GID '0' already exists
Adding enduser to local docker group
usermod: group 'docker' does not exist
```
doesn't seem to _break_ it, but we should think through this one before merging

**Comment by rbren** [comment]
See also: https://github.com/OpenDevin/OpenDevin/pull/1651

**Comment by rbren** [comment]
@imkebe can you try running with tag `1651-merge`? I just incorporated some of your changes here into that PR

**Comment by imkebe** [comment]
@rbren group issue is solved. There is now another `source ~/.bash.rc` but it's out of scope of this PR. 
The `0` group id means that it's root. It seems not to be the way docker should be handled
However at the end we are creating the group inside container and root group will be there and will have all permissions so it' should be a huge problem.

File: containers/app/entrypoint.sh (hunk 1 above)
Comment (SmartManoj) [reply containers/app/entrypoint.sh:38]: yes, but this won't break the flow.

**Comment by SmartManoj** [comment]
> I think we need to be a little more careful here.
> 
> This is what I get when running:
> 
> ```
> Creating docker group with fetched gid.
> groupadd: GID '0' already exists
> Adding enduser to local docker group
> usermod: group 'docker' does not exist
> ```
> 
> doesn't seem to _break_ it, but we should think through this one before merging

As these statements are used to run docker as non-root, running as root won't break it.
https://docs.docker.com/engine/install/linux-postinstall/

**Comment by wmsmigiel** [comment]
> @imkebe can you try running with tag `1651-merge`? I just incorporated some of your changes here into that PR

this fixed the docker issue for me (see #1685) but litellm pricing problem persist (#1667)

File: containers/app/entrypoint.sh (hunk 1 above)
Comment (SmartManoj) [reply containers/app/entrypoint.sh:38]: yes, but it doesn't affect the flow.

**Comment by rbren** [comment]
Hey all--I think all these issues (including the bashrc timeout) have been solved at this point

Thanks @imkebe for the inspiration here--it was a big help in tracking down one of the major issues!

**Pull request closed without merging**